startup.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import sys
+import re
+import glob
+
+from sgtk.platform import SoftwareLauncher, SoftwareVersion, LaunchInformation
+
+class PhotoshopLauncher(SoftwareLauncher):
+    """
+    Handles the launching of Photoshop. Contains the logic for
+    scanning for installed versions of the software and
+    how to correctly set up a launch environment for the tk-photoshopcc
+    engine.
+    """
+
+    def scan_software(self, versions=None):
+        """
+        Performs a scan for software installations.
+
+        :param list versions: List of strings representing versions
+                              to search for. If set to None, search
+                              for all versions. A version string is
+                              DCC-specific but could be something
+                              like "2017", "6.3v7" or "1.2.3.52".
+        :returns: List of :class:`SoftwareVersion` instances
+        """
+        icon_path = os.path.join(self.disk_location, "resources", "ps_2017_icon_256.png")
+
+        if sys.platform == "darwin":
+            # Default installs are located here:
+            # /Applications/Adobe Photoshop CC 2017/Adobe Photoshop CC 2017.app
+            glob_pattern = "/Applications/Adobe Photoshop CC [0-9][0-9][0-9][0-9]/Adobe Photoshop CC [0-9][0-9][0-9][0-9].app"
+            version_regex = re.compile(
+                "^/Applications/Adobe Photoshop CC ([0-9]{4})/Adobe Photoshop CC ([0-9]{4})\.app$",
```

Comment:
I don't think this will catch `2015.5`. I'm thinking we should use `*` in globs and `.*` in regex as much as possible to account for the whims of the 3rd party versioning schemes? What do you think?